components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListener.java

```diff
@@ -282,4 +294,79 @@ private void removeEntriesFromCache(ServiceProvider serviceProvider, String tena
         }
 
     }
+
+    /**
+     * Retrieves the value of SaaS property before application is updated
+     *
+     * @param serviceProvider ServiceProvider
+     * @param tenantDomain    String
+     * @throws IdentityApplicationManagementException
+     */
+    private void getSaaSPropertyValue(ServiceProvider serviceProvider, String tenantDomain) throws IdentityApplicationManagementException {
+
+        ServiceProvider sp = OAuth2ServiceComponentHolder.getApplicationMgtService()
+                .getServiceProvider(serviceProvider.getApplicationName(), tenantDomain);
+        IdentityUtil.threadLocalProperties.get().put(SAAS_PROPERTY, sp.isSaasApp());
+    }
+
+    /**
+     * validates tenant domain and revokes access tokens of OAuth application if SaaS is disabled
+     *
+     * @param serviceProvider ServiceProvider
+     * @param tenantDomain    String
+     * @throws IdentityApplicationManagementException
+     */
+    private void validateTenantDomain(ServiceProvider serviceProvider, String tenantDomain) throws IdentityApplicationManagementException {
+
+        TokenMgtDAO tokenMgtDAO = new TokenMgtDAO();
+        boolean saasEnabled = false;
+        Object saasStatus = IdentityUtil.threadLocalProperties.get().get(SAAS_PROPERTY);
+        if (saasStatus != null) {
+            saasEnabled = (Boolean) saasStatus;
+        }
+        if ((saasEnabled == true) && (serviceProvider.isSaasApp() == false)) {
```

Comment:
'== true' is redundant. "serviceProvider.isSaasApp() == false" also can be reduced to "!serviceProvider.isSaasApp()"